lib/src/modules/task.cc

```diff
@@ -592,6 +706,20 @@ ActionResponse Task::callAction(const ActionRequest& request)
 
     task_params.set<std::string>("_task", task_name);
 
+    std::vector<std::string> meta_files;
+    // if metadata::files key not available catch error
+    try {
+         meta_files = task_metadata.getWithDefault<std::vector<std::string>>("files", {});
```

Comment:
Decided not to rework this part? Maybe instead guard it with `task_metadata != task_execution_params`.

Reply:
The comment doesn't feel clear to me either.

Reply:
I am planning on re-working this. I need to understand what information is included in `request.params()` to avoid defaulting to `task_execution_params` when initializing `task_metadata`.